python/GafferOSL/FocalBlur.gfr

```diff
@@ -0,0 +1,889 @@
+__children["FocalBlur"]["SceneCameraExpression"]["__in"]["p6"].setInput( __children["FocalBlur"]["CameraQuery"]["out"][3]["source"] )
+__children["FocalBlur"]["SceneCameraExpression"]["__in"]["p7"].setInput( __children["FocalBlur"]["CameraQuery"]["out"][3]["value"] )
+__children["FocalBlur"]["SceneCameraExpression"]["__in"]["p8"].setInput( __children["FocalBlur"]["CameraQuery"]["out"][4]["source"] )
+__children["FocalBlur"]["SceneCameraExpression"]["__in"]["p9"].setInput( __children["FocalBlur"]["CameraQuery"]["out"][4]["value"] )
+__children["FocalBlur"]["SceneCameraExpression"]["__uiPosition"].setValue( imath.V2f( 768.393555, 21.7875843 ) )
+__children["FocalBlur"]["CameraQuery"]["scene"].setInput( __children["FocalBlur"]["BoxIn2"]["out"] )
+__children["FocalBlur"]["CameraQuery"]["cameraMode"].setInput( __children["FocalBlur"]["Expression"]["__out"]["p0"] )
+__children["FocalBlur"]["CameraQuery"]["location"].setInput( __children["FocalBlur"]["cameraPath"] )
+__children["FocalBlur"]["CameraQuery"]["queries"][0].setValue( 'focalLength' )
+__children["FocalBlur"]["CameraQuery"]["queries"][1].setValue( 'aperture' )
+__children["FocalBlur"]["CameraQuery"]["queries"][2].setValue( 'focusDistance' )
+__children["FocalBlur"]["CameraQuery"]["queries"][3].setValue( 'fStop' )
+__children["FocalBlur"]["CameraQuery"]["queries"][4].setValue( 'focalLengthWorldScale' )
+__children["FocalBlur"]["CameraQuery"]["__uiPosition"].setValue( imath.V2f( 767.942566, 29.8591805 ) )
+__children["FocalBlur"]["CustomCameraExpression"]["__in"]["p0"].setInput( __children["FocalBlur"]["fStop"] )
+__children["FocalBlur"]["CustomCameraExpression"]["__in"]["p1"].setInput( __children["FocalBlur"]["focalLength"] )
+__children["FocalBlur"]["CustomCameraExpression"]["__in"]["p2"].setInput( __children["FocalBlur"]["focalLengthWorldScale"] )
+__children["FocalBlur"]["CustomCameraExpression"]["__in"]["p3"].setInput( __children["FocalBlur"]["focusDistance"] )
+__children["FocalBlur"]["CustomCameraExpression"]["__in"]["p4"].setInput( __children["FocalBlur"]["horizontalAperture"] )
+__children["FocalBlur"]["CustomCameraExpression"]["__uiPosition"].setValue( imath.V2f( 765.393494, 21.787466 ) )
+__children["FocalBlur"]["NoCameraExpression"]["__in"]["p0"].setInput( __children["FocalBlur"]["focusDistance"] )
+__children["FocalBlur"]["NoCameraExpression"]["__uiPosition"].setValue( imath.V2f( 762.390442, 21.7854328 ) )
+__children["FocalBlur"]["Dot25"]["in"].setInput( __children["FocalBlur"]["Dot12"]["out"] )
+__children["FocalBlur"]["Dot25"]["__uiPosition"].setValue( imath.V2f( 822.9505, 85.5897293 ) )
+__children["FocalBlur"]["Dot26"]["in"].setInput( __children["FocalBlur"]["Dot25"]["out"] )
+__children["FocalBlur"]["Dot26"]["__uiPosition"].setValue( imath.V2f( 822.9505, 35.2791519 ) )
+Gaffer.Metadata.registerValue( __children["FocalBlur"]["cameraPath"], 'nodule:type', '' )
```

Comment:
Maybe we should use a `GafferSceneUI.ScenePathPlugValueWidget` here so users can browse for the camera? I think this may throw a small spanner as far as the placeholderText goes...

Reply:
I've made this change, though, yes, it does throw a spanner in the works as far as placeholderText is concerned.